tests/unit/test_connection/test_request.py

```diff
@@ -30,42 +31,47 @@ def _route_handler() -> None:
     pass
 
 
-async def test_request_empty_body_to_json(anyio_backend: str) -> None:
+@pytest.fixture(name="scope")
+def scope_fixture(create_scope: Callable[..., Scope]) -> Scope:
+    return create_scope(type="http", route_handler=_route_handler)
+
+
+async def test_request_empty_body_to_json(anyio_backend: str, scope: Scope) -> None:
     with patch.object(Request, "body", return_value=b""):
-        request_empty_payload: Request = Request(scope={"type": "http", "route_handler": _route_handler})  # type: ignore
```

Comment:
Now that we aren't caching connection things in the top-level of `scope`, when we create connection instances, we need to ensure that the scope that is passed to the connection has the "state" key in it, like it would at runtime.

That means this change has had to cover a bit more surface area to accommodate this, but I think it is a better pattern overall to use the `create_scope` fixture in our tests anyway.